src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/TypeLoaderEnvironment.ConstructedGenericMethodsLookup.cs

```diff
@@ -293,14 +293,14 @@ public bool TryGetGenericMethodComponents(IntPtr methodDictionary, out RuntimeTy
             return true;
         }
 
-        public bool TryLookupExactMethodPointerForComponents(RuntimeTypeHandle declaringType, MethodNameAndSignature nameAndSignature, RuntimeTypeHandle[] genericMethodArgumentHandles, out IntPtr result)
+        public bool TryLookupExactMethodPointerForComponents(InstantiatedMethod method, out IntPtr result)
         {
-            int lookupHashcode = declaringType.GetHashCode();
+            int lookupHashcode = method.OwningType.GetHashCode();
 
             NativeHashtable hashtable;
             ExternalReferencesTable externalReferencesLookup;
 
-            HandleBasedGenericMethodLookup lookupData = new HandleBasedGenericMethodLookup(declaringType, nameAndSignature, genericMethodArgumentHandles);
```

Comment:
Delete HandleBasedGenericMethodLookup? There is only one other use of it that can be trivially switched over to MethodDescBasedGenericMethodLookup.